src/vs/workbench/contrib/positronAssistant/browser/languageModelModalDialog.tsx

```diff
@@ -338,39 +377,43 @@ const LanguageModelConfiguration = (props: React.PropsWithChildren<LanguageModel
 								displayName={provider.options.title ?? provider.options.identifier}
 								identifier={provider.options.identifier}
 								selected={provider.options.identifier === source.provider.id}
-								onClick={() => setSource(provider.options.value)}
+								onClick={() => {
+									setSource(provider.options.value);
+									setProviderConfig({ ...provider.options.value, ...provider.options.value.defaults, provider: provider.options.identifier });
+									setError(undefined);
+								}}
 							/>
 						})
 					}
 				</div>
 				<label className='language-model-section'>
 					{(() => localize('positron.newConnectionModalDialog.authentication', "Authentication"))()}
 				</label>
-				{source?.supportedOptions.includes('apiKey') &&
-					(
-						<div className='language-model-authentication-container' id='api-key-input'>
-							<LabeledTextInput
-								label={(() => localize('positron.newConnectionModalDialog.apiKey', "API Key"))()}
-								type='password'
-								validator={(value) => {
-									if (errorMessage) {
-										return errorMessage;
-									}
-									return value ? undefined : localize('positron.newConnectionModalDialog.missingApiKey', 'An API key is required')
-								}}
-								value={apiKey ?? ''}
-								onChange={e => { setApiKey(e.currentTarget.value) }}
-							/>
-							{signInButton()}
-						</div>
-					)
-				}
-				{!source?.supportedOptions.includes('apiKey') &&
-					signInButton()
-				}
+				<div className='language-model-authentication-method-container'>
+					<RadioGroup
+						entries={authMethodRadioButtons}
+						initialSelectionId='apiKey'
+						name='authMethod'
+						onSelectionChanged={(authMethod) => {
+							// setAuthMethod(authMethod);
+							console.log(authMethod);
+						}}
```

Comment:
is this still needed?

Reply:
Good point, I should remove the console logging